From 3795a011eeb9730cda9f0beadce70aab8aa71e68 Mon Sep 17 00:00:00 2001
From: Paul Floyd <pjfloyd@wanadoo.fr>
Date: Sat, 4 May 2024 15:03:11 +0200
Subject: [PATCH 04/11] Bug 486569 - linux inotify_init syscall wrapper missing
 POST entry in syscall_table

(cherry picked from commit b38115cc6087f30f872c533f93c7c31a6d73eb24)
---
 NEWS                                       | 1 +
 coregrind/m_syswrap/syswrap-amd64-linux.c  | 2 +-
 coregrind/m_syswrap/syswrap-arm-linux.c    | 2 +-
 coregrind/m_syswrap/syswrap-mips32-linux.c | 2 +-
 coregrind/m_syswrap/syswrap-mips64-linux.c | 2 +-
 coregrind/m_syswrap/syswrap-ppc32-linux.c  | 2 +-
 coregrind/m_syswrap/syswrap-ppc64-linux.c  | 2 +-
 coregrind/m_syswrap/syswrap-s390x-linux.c  | 2 +-
 coregrind/m_syswrap/syswrap-x86-linux.c    | 2 +-
 9 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/NEWS b/NEWS
index f674191a286a..b65f9206679b 100644
--- a/NEWS
+++ b/NEWS
@@ -7,6 +7,7 @@ The following bugs have been fixed or resolved on this branch.
 
 486180  [MIPS] 'VexGuestArchState' has no member named 'guest_IP_AT_SYSCALL'
 486293  memccpy false positives
+486569  linux inotify_init syscall wrapper missing POST entry in syscall_table
 
 To see details of a given bug, visit
   https://bugs.kde.org/show_bug.cgi?id=XXXXXX
diff --git a/coregrind/m_syswrap/syswrap-amd64-linux.c b/coregrind/m_syswrap/syswrap-amd64-linux.c
index a59e01826ca2..86a47d9c2811 100644
--- a/coregrind/m_syswrap/syswrap-amd64-linux.c
+++ b/coregrind/m_syswrap/syswrap-amd64-linux.c
@@ -773,7 +773,7 @@ static SyscallTableEntry syscall_table[] = {
    LINXY(__NR_keyctl,            sys_keyctl),         // 250
    LINX_(__NR_ioprio_set,        sys_ioprio_set),     // 251
    LINX_(__NR_ioprio_get,        sys_ioprio_get),     // 252
-   LINX_(__NR_inotify_init,	 sys_inotify_init),   // 253
+   LINXY(__NR_inotify_init,	 sys_inotify_init),   // 253
    LINX_(__NR_inotify_add_watch, sys_inotify_add_watch), // 254
 
    LINX_(__NR_inotify_rm_watch,	 sys_inotify_rm_watch), // 255
diff --git a/coregrind/m_syswrap/syswrap-arm-linux.c b/coregrind/m_syswrap/syswrap-arm-linux.c
index 217b1c49dc9d..7b536e52cb2e 100644
--- a/coregrind/m_syswrap/syswrap-arm-linux.c
+++ b/coregrind/m_syswrap/syswrap-arm-linux.c
@@ -922,7 +922,7 @@ static SyscallTableEntry syscall_main_table[] = {
 //   LINX_(__NR_ioprio_set,        sys_ioprio_set),     // 289
 
 //   LINX_(__NR_ioprio_get,        sys_ioprio_get),     // 290
-   LINX_(__NR_inotify_init,    sys_inotify_init),   // 291
+   LINXY(__NR_inotify_init,    sys_inotify_init),   // 291
    LINX_(__NR_inotify_add_watch, sys_inotify_add_watch), // 292
    LINX_(__NR_inotify_rm_watch,    sys_inotify_rm_watch), // 293
 //   LINX_(__NR_migrate_pages,    sys_migrate_pages),    // 294
diff --git a/coregrind/m_syswrap/syswrap-mips32-linux.c b/coregrind/m_syswrap/syswrap-mips32-linux.c
index 3f991da0a18d..e2f499eb80f8 100644
--- a/coregrind/m_syswrap/syswrap-mips32-linux.c
+++ b/coregrind/m_syswrap/syswrap-mips32-linux.c
@@ -1037,7 +1037,7 @@ static SyscallTableEntry syscall_main_table[] = {
    LINXY (__NR_mq_timedreceive,        sys_mq_timedreceive),         // 274
    LINX_ (__NR_mq_notify,              sys_mq_notify),               // 275
    LINXY (__NR_mq_getsetattr,          sys_mq_getsetattr),           // 276
-   LINX_ (__NR_inotify_init,           sys_inotify_init),            // 275
+   LINXY (__NR_inotify_init,           sys_inotify_init),            // 275
    LINX_ (__NR_inotify_add_watch,      sys_inotify_add_watch),       // 276
    LINX_ (__NR_inotify_rm_watch,       sys_inotify_rm_watch),        // 277
    LINXY (__NR_waitid,                 sys_waitid),                  // 278
diff --git a/coregrind/m_syswrap/syswrap-mips64-linux.c b/coregrind/m_syswrap/syswrap-mips64-linux.c
index 9899a21cfa78..24f2bcc93956 100644
--- a/coregrind/m_syswrap/syswrap-mips64-linux.c
+++ b/coregrind/m_syswrap/syswrap-mips64-linux.c
@@ -759,7 +759,7 @@ static SyscallTableEntry syscall_main_table[] = {
    LINX_ (__NR_request_key, sys_request_key),
    LINXY (__NR_keyctl, sys_keyctl),
    PLAX_ (__NR_set_thread_area, sys_set_thread_area),
-   LINX_ (__NR_inotify_init, sys_inotify_init),
+   LINXY (__NR_inotify_init, sys_inotify_init),
    LINX_ (__NR_inotify_add_watch, sys_inotify_add_watch),
    LINX_ (__NR_inotify_rm_watch, sys_inotify_rm_watch),
    LINXY (__NR_openat, sys_openat),
diff --git a/coregrind/m_syswrap/syswrap-ppc32-linux.c b/coregrind/m_syswrap/syswrap-ppc32-linux.c
index 0aabfbb179ec..74ba68564cd8 100644
--- a/coregrind/m_syswrap/syswrap-ppc32-linux.c
+++ b/coregrind/m_syswrap/syswrap-ppc32-linux.c
@@ -944,7 +944,7 @@ static SyscallTableEntry syscall_table[] = {
    LINX_(__NR_ioprio_set,        sys_ioprio_set),         // 273
    LINX_(__NR_ioprio_get,        sys_ioprio_get),         // 274
 
-   LINX_(__NR_inotify_init,  sys_inotify_init),               // 275
+   LINXY(__NR_inotify_init,  sys_inotify_init),               // 275
    LINX_(__NR_inotify_add_watch,  sys_inotify_add_watch),     // 276
    LINX_(__NR_inotify_rm_watch,   sys_inotify_rm_watch),      // 277
    PLAXY(__NR_spu_run,            sys_spu_run),               // 278
diff --git a/coregrind/m_syswrap/syswrap-ppc64-linux.c b/coregrind/m_syswrap/syswrap-ppc64-linux.c
index 35e3f8ec4ee4..4609e10df363 100644
--- a/coregrind/m_syswrap/syswrap-ppc64-linux.c
+++ b/coregrind/m_syswrap/syswrap-ppc64-linux.c
@@ -928,7 +928,7 @@ static SyscallTableEntry syscall_table[] = {
    LINX_(__NR_ioprio_set,        sys_ioprio_set),         // 273
    LINX_(__NR_ioprio_get,        sys_ioprio_get),         // 274
 
-   LINX_(__NR_inotify_init,  sys_inotify_init),           // 275
+   LINXY(__NR_inotify_init,  sys_inotify_init),           // 275
    LINX_(__NR_inotify_add_watch,  sys_inotify_add_watch), // 276
    LINX_(__NR_inotify_rm_watch,   sys_inotify_rm_watch),  // 277
 
diff --git a/coregrind/m_syswrap/syswrap-s390x-linux.c b/coregrind/m_syswrap/syswrap-s390x-linux.c
index f941bdd1907a..9ed2fa46888f 100644
--- a/coregrind/m_syswrap/syswrap-s390x-linux.c
+++ b/coregrind/m_syswrap/syswrap-s390x-linux.c
@@ -751,7 +751,7 @@ static SyscallTableEntry syscall_table[] = {
    LINXY(__NR_waitid, sys_waitid),                                    // 281
    LINX_(__NR_ioprio_set,  sys_ioprio_set),                           // 282
    LINX_(__NR_ioprio_get,  sys_ioprio_get),                           // 283
-   LINX_(__NR_inotify_init,  sys_inotify_init),                       // 284
+   LINXY(__NR_inotify_init,  sys_inotify_init),                       // 284
 
    LINX_(__NR_inotify_add_watch,  sys_inotify_add_watch),             // 285
    LINX_(__NR_inotify_rm_watch,  sys_inotify_rm_watch),               // 286
diff --git a/coregrind/m_syswrap/syswrap-x86-linux.c b/coregrind/m_syswrap/syswrap-x86-linux.c
index c6cf682e797c..ea930bafe3ec 100644
--- a/coregrind/m_syswrap/syswrap-x86-linux.c
+++ b/coregrind/m_syswrap/syswrap-x86-linux.c
@@ -1508,7 +1508,7 @@ static SyscallTableEntry syscall_table[] = {
    LINX_(__NR_ioprio_set,        sys_ioprio_set),     // 289
 
    LINX_(__NR_ioprio_get,        sys_ioprio_get),     // 290
-   LINX_(__NR_inotify_init,	 sys_inotify_init),   // 291
+   LINXY(__NR_inotify_init,	 sys_inotify_init),   // 291
    LINX_(__NR_inotify_add_watch, sys_inotify_add_watch), // 292
    LINX_(__NR_inotify_rm_watch,	 sys_inotify_rm_watch), // 293
 //   LINX_(__NR_migrate_pages,	 sys_migrate_pages),    // 294
-- 
2.45.2

